### PR TITLE
workflows: invoke self-hosted logic on Linux

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Hopefully fixes the errors we get on it.

### CODE REVIEW

**Comment by Bo98** [comment]
Ah, I see dealing with file ownership is still an absolute nightmare in Docker.

File: Formula/hello.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# test
```
Comment (MikeMcQuaid) [inline Formula/hello.rb:1]:
```suggestion
```

File: .github/workflows/dispatch-build-bottle.yml [hunk 2]
```diff
@@ -73,10 +73,9 @@ jobs:
         run: |
           if [ "$RUNNER_OS" = 'macOS' ]; then
             echo 'PATH=/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin' >> $GITHUB_ENV
-            echo 'GITHUB_ACTIONS_HOMEBREW_MACOS_SELF_HOSTED=1' >> $GITHUB_ENV
-          else
-            # No PATH needed on Linux as set by Docker
-            echo 'HOMEBREW_FORCE_HOMEBREW_ON_LINUX=1' >> $GITHUB_ENV
+          fi
+          if [ "${{matrix.runner != 'ubuntu-latest'}}" = "true" ]; then
```
Comment (MikeMcQuaid) [inline .github/workflows/dispatch-build-bottle.yml:77]: Complete pedantry but I wonder if we want to handle the case where we might ever use `macOS-latest` or pinned Ubuntu versions here? Ideally there'd be a way of actually detecting this not based on name 🤔. Don't block this PR on that, though.

**Review by MikeMcQuaid — DISMISSED** [review]
Thanks @Bo98!

File: .github/workflows/dispatch-build-bottle.yml (hunk 2 above)
Comment (dawidd6) [reply .github/workflows/dispatch-build-bottle.yml:77]: It could easily be done based on label and runner os.

**Comment by Bo98** [comment]
Here's another thought: would it work if we [add an `.env` file](https://docs.github.com/en/actions/hosting-your-own-runners/using-a-proxy-server-with-self-hosted-runners#using-a-env-file-to-set-the-proxy-configuration) to each of our self-hosted runners with `GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED=1`?

**Comment by dawidd6** [comment]
That would be great IMO.

**Review by dawidd6 — APPROVED** [review]

**Comment by Bo98** [comment]
Old PRs will still break things unless they rebase upon this commit, but this hopefully fix everything going forward.

**Pull request merged**